src/index.ts

```diff
@@ -20,6 +20,78 @@ export type RequestBodyType = {
   ed?: string
 }
 
+// Should we make EcommerceType part of ComponentSettings? It will be used on every MC with ecommerce support
+type EcommerceType = {
+  order_id: number | string
+  currency: string
+  revenue: number | string
+  total: number | string
+  value: number | string
+  quantity: number | string
+  products: Product[] | null
+  checkout_id: number | string
+  affiliation: string
+  shipping: number | string
+  tax: number | string
+  discount: number | string
+  coupon: string
+  creative: string
+  query: string
+  step: number | string
+  payment_type: string
+}
+
+// Should we make Product part of ComponentSettings? It will be used on every MC with ecommerce support
+export type Product = {
+  product_id: number | string
+  sku: number | string
+  name: string
+  category: string
+  brand: string
+  price: number | string
+  quantity: number
+  variant: string
+  currency: string
+  value: number | string
+  position: number | string
+  coupon: number | string
+}
+
+function mapEcommerceData(
+  ecommerce: EcommerceType
+): Record<string, string | number> | null {
+  const transformedProductData: Record<string, string | number> = {}
+  if (!ecommerce || !ecommerce.products) {
+    return null
+  } else {
+    ecommerce.products.forEach((product, index) =>
+      [
+        'product_id',
+        'sku',
+        'category',
+        'name',
+        'brand',
+        'variant',
+        'price',
+      ].forEach(prop => {
+        const key = `product_${prop}[${index}]`
+        transformedProductData[key] =
+          product[prop as keyof Product] || product.sku
+      })
+    )
+  }
+
+  const ecommerceData = {
+    order_id: ecommerce.order_id,
+    currency: ecommerce.currency,
+    value: ecommerce.revenue || ecommerce.total || ecommerce.value,
```

Comment:
A few questions regarding this line:

1. What is the reason for taking one of those attributes and discarding the others? Is it really our intention here to skip the `value` attribute if for example the `revenue` attribute is present?

2. If `revenue` *is* set, but to exactly the number `0`, the `revenue` attribute is going to be discarded, and `total` or `value` used instead (because `0` is falsy). Is that the desired behavior?